community/duo_unix: improve aport

ktreese opened the pull request:
https://duo.com/docs/duounix
Duo two-factor authentication for Unix systems with Pluggable Authentication Modules (PAM)

rnalrd commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,38 @@
+pkgdesc="Duo two-factor authentication for Unix systems with Pluggable Authentication Modules (PAM)"
+url="https://duo.com/docs/duounix"
 arch="all"
-license="GPL-2.0-only"
```
Looks a valid license ID.

ktreese replied:
Updated to `GPL-2.0` based on https://spdx.org/licenses/

rnalrd commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,38 @@
-depends=""
-depends_dev="openssl-dev zlib-dev"
-makedepends="$depends_dev automake autoconf libtool"
+license="GPL"
```
List of valid IDs: https://duo.com/support/documentation/duounix

rnalrd commented:
Why creating clone in testing?

rnalrd commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,38 @@
 	default_prepare
-	./bootstrap
 }
+
```
This function can be removed altogether. There's nothing custom defined here.

ktreese replied:
Resolved via be68e705ba

rnalrd commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,38 @@
+check() {
+	cd "$builddir"
+}
+
```
This function does effectively nothing. If a test suite does not exist add

"options=!check" # no test suite

among the APKBUILD variables at the beginning. Otherwise please run the tests here.

ktreese replied:
Resolved via be68e705ba

rnalrd commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,38 @@
 
 package() {
 	cd "$builddir"
+	mkdir -p "$pkgdir"
```
abuild creates it for you.

ktreese replied:
Resolved via be68e705ba

rnalrd commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,38 @@
 }
+
 build() {
 	cd "$builddir"
```
Nowdays it can be omitted. abuild does it for you.

ktreese replied:
Resolved via be68e705ba

rnalrd commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,38 @@
-depends=""
-depends_dev="openssl-dev zlib-dev"
-makedepends="$depends_dev automake autoconf libtool"
+license="GPL-2.0"
```
I see that is GPL-2.0 plus a custom clause, so I guess it's more fitting than "GPL-2.0-only".

ktreese replied:
works for me

ktreese commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,38 @@
 }
 
 package() {
 	cd "$builddir"
```
Resolved via be68e705ba

andypost commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,27 @@
+license="GPL-2.0"
+depends="openssh openssh-server-pam"
+makedepends="openssl-dev linux-pam-dev openssh-server-pam libtool"
+install=""
```
No reason in empty line

ktreese replied:
Resolved via ad4c5bf9df

andypost commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,27 @@
-source="https://github.com/duosecurity/duo_unix/archive/$pkgname-$pkgver.tar.gz"
-builddir="$srcdir"/$pkgname-$pkgname-$pkgver
+license="GPL-2.0"
+depends="openssh openssh-server-pam"
```
OS-pam should not be duplicated in makedepends

ktreese replied:
Resolved in 3819dac5bf

andypost commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,27 @@
+subpackages="$pkgname-doc"
+source="https://dl.duosecurity.com/$pkgname-$pkgver.tar.gz"
+builddir="$srcdir/$pkgname-$pkgver"
+options="suid !check"
```
Why no check? It needs comment or add check, basically to make sure that package works

ktreese replied:
Resolved via fb84c2c325

ktreese replied:
Readded `!check` as check consistently fails with:

```
In file included from /usr/include/fortify/stdio.h:20:0,
                 from testpam_preload.c:16:
testpam_preload.c:107:1: error: redefinition of 'fopen'
 fopen64(const char *filename, const char *mode)
 ^
/usr/include/stdio.h:63:7: note: previous definition of 'fopen' was here
 FILE *fopen(const char *__restrict, const char *__restrict);
       ^~~~~
make[2]: *** [Makefile:463: testpam_preload.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/builder/duo_unix/src/duo_unix-1.11.1/tests'
make[1]: *** [Makefile:652: check-am] Error 2
make[1]: Leaving directory '/home/builder/duo_unix/src/duo_unix-1.11.1/tests'
make: *** [Makefile:469: check-recursive] Error 1
make: Leaving directory '/home/builder/duo_unix/src/duo_unix-1.11.1'
>>> ERROR: duo_unix*: check failed
>>> ERROR: duo_unix: all failed
```

Any suggestions on how to resolve this?

andypost commented on community/duo_unix/APKBUILD:
```diff
@@ -1,39 +1,27 @@
-# Contributor: Alan Lacerda <alacerda@alpinelinux.org>
-# Maintainer: Paul Morgan <jumanjiman@gmail.com>
+# Contributor: Kris Reese <ktreese@gmail.com>
+# Maintainer: Kris Reese <ktreese@gmail.com>
```
Not sure you need to remove previous contribution

ktreese replied:
Oversight on my part.  Corrected in 738f14a203

Ikke commented:
@andypost Can you elaborate what you meant with that last comment?

Ikke commented:
@ktreese Maybe you can split this up into two commits. One bumping the version and one doing a clean up.

ktreese commented:
@Ikke let me know if this is what you had in mind.  Thanks!

Ikke commented:
@ktreese Sorry, I forgot to respond.

This already looks better. Just some improvements.

543100dd7c7ce60c51b0cd9479fbbc6025a93bde says new aport, while you seem to improve on an existing aport. The commit message mentions all kinds of details that can be inferred from the diff already. A more useful commit message explains why things are being done (but it's not necessary to explain every detail either).

It also resets `pkgrel` without the version being updated (that's only done in a commit later).

The changes look good to me, so if you could fix that, I will push this.

Thanks.

ktreese commented:
no problem @Ikke!  I believe I've fixed up the things requested of me.  Let me know if I've overlooked anything else!  The reason the commit message was verbose was because I was keeping them as I rebased.  They should all be cleared out now!  Thank you.
